packages/client/src/index.ts

```diff
@@ -41,7 +60,22 @@ export interface RoutingV1HttpApiClient {
    * Returns an async generator of PeerInfos that can provide the content
    * for the passed CID
    */
-  getProviders(cid: CID, options?: AbortOptions): AsyncGenerator<PeerInfo>
+  getProviders(cid: CID, options?: AbortOptions): AsyncGenerator<Record>
```

Comment:
Can the return type be guaranteed to be `BitswapRecord`?

Reply:
@achingbrain we don't want BitswapRecords to be used, see https://specs.ipfs.tech/routing/http-routing-v1/#get-routing-v1-providers-cid - it should be a generic record. PeerRecords are recommended.

Reply:
Then can the return type be guaranteed to be `PeerRecord`?  What I'm getting at here is it makes the users life easier if they don't have to deal with a union type.

Reply:
Formally it could have custom schema, but in practice you can ignore unknown schemas and parse bitswap schema as peer schema. 

For the wider context, the only `/routing/v1` implementation (afaik) that returned [bitswap schema](https://specs.ipfs.tech/routing/http-routing-v1/#bitswap-schema) was cid.contact (IPNI), and it had `ID` and `Addrs` just like [peer schema](https://specs.ipfs.tech/routing/http-routing-v1/#peer-schema) does.

Reply:
Ok, sounds like we can just convert any `BitswapRecord`s to `PeerRecord`s and drop the union type then.